rpc/src/types/blockchain.rs

```diff
@@ -0,0 +1,455 @@
+use crate::types::proposal_short_id::ProposalShortId;
+use crate::types::Bytes;
+use ckb_core::block::{Block as CoreBlock, BlockBuilder};
+use ckb_core::header::{Header as CoreHeader, HeaderBuilder, Seal as CoreSeal};
+use ckb_core::script::Script as CoreScript;
+use ckb_core::transaction::{
+    CellInput as CoreCellInput, CellOutput as CoreCellOutput, OutPoint as CoreOutPoint,
+    Transaction as CoreTransaction, TransactionBuilder,
+};
+use ckb_core::uncle::UncleBlock as CoreUncleBlock;
+use ckb_core::{BlockNumber, Capacity};
+use numext_fixed_hash::H256;
+use numext_fixed_uint::U256;
+use serde_derive::{Deserialize, Serialize};
+
+#[derive(Clone, Default, Serialize, Deserialize, PartialEq, Eq, Hash, Debug)]
+pub struct Script {
+    pub(crate) version: u8,
+    pub(crate) args: Vec<Bytes>,
+    pub(crate) reference: Option<H256>,
+    pub(crate) binary: Option<Bytes>,
+    pub(crate) signed_args: Vec<Bytes>,
+}
+
+impl From<Script> for CoreScript {
+    fn from(json: Script) -> CoreScript {
+        let Script {
+            version,
+            args,
+            reference,
+            binary,
+            signed_args,
+        } = json;
+        CoreScript::new(
+            version,
+            args.into_iter().map(|arg| arg.into_vec()).collect(),
+            reference,
+            binary.map(|b| b.into_vec()),
+            signed_args.into_iter().map(|arg| arg.into_vec()).collect(),
+        )
+    }
+}
+
+impl From<CoreScript> for Script {
+    fn from(core: CoreScript) -> Script {
+        let (version, args, reference, binary, signed_args) = core.destruct();
```

Comment:
`destruct` is only called in `From` and the data is already consumed by borrowship, maybe we can remove `destruct` method?

Reply:
Core types do not expose field directly, so you can't and shouldn't just destruct outside such as 
`let CoreScript{ version, args, reference, binary, signed_args } = core;`
`destruct` are a temporary solution, for now. If we decide to implement zero-copy types in the future, they may change.